CRM/Price/BAO/PriceSet.php

```diff
@@ -871,13 +871,13 @@ public static function buildPriceSet(&$form) {
     }
 
     if ($className == 'CRM_Contribute_Form_Contribution_Main') {
-      $feeBlock = &$form->_values['fee'];
+      $feeBlock = $form->_values['fee'];
       if (!empty($form->_useForMember)) {
         $component = 'membership';
       }
     }
     else {
-      $feeBlock = &$form->_priceSet['fields'];
+      $feeBlock = $form->_priceSet['fields'];
```

Comment:
I haven't traced this but if it helps - just looking at the code below if you take away the `&` here then the hook won't be able to update $feeBlock, so when it's passed to addPriceFieldsToForm it's still the original value.

Reply:
Correction: Won't be able to update the original `$form->_priceSet['fields']`.

Reply:
Ah yes, thankyou. I just saw the `&` and thought PHP4..

Reply:
Removed from this PR